### PR TITLE
Use Lighthouse Types instead of maintaining our own

### PR DESCRIPTION
**Pull request by ec2**
Notes:
- Prover RPC API changed to now accept ForkName as a param so we can deserialize LightClientUpdates according to which fork were on.
- Rotation sepolia tests are failing in preprocessor (I have a feeling we can remove this test entirely and rely on `test_both_circuit_sepolia`)

TODO:
~I am kinda annoyed that we have Lighthouses EthSpec and then our Spec trait. There is a huge overlap, so currently thinking how we can simplify this.~ Done

### CODE REVIEW

File: lightclient-circuits/src/committee_update_circuit.rs [hunk 1]
```diff
@@ -29,8 +29,9 @@ use halo2_ecc::{
     fields::FieldChip,
 };
 use itertools::Itertools;
-use ssz_rs::Merkleized;
+// use ssz_rs::Merkleized;
```
Comment (nulltea) [inline lightclient-circuits/src/committee_update_circuit.rs]:
```suggestion
```

File: lightclient-circuits/src/witness/multiproof.rs [hunk 2]
```diff
@@ -178,7 +178,10 @@ pub fn merkle_tree(leaves: &[Node]) -> Vec<Node> {
     o
 }
 
-pub fn create_multiproof(merkle_tree: &[Node], indices_to_prove: &[GeneralizedIndex]) -> Vec<Node> {
+pub fn create_multiproof(
```
Comment (nulltea) [inline lightclient-circuits/src/witness/multiproof.rs:181]: multiproof stuff stays?

File: preprocessor/Cargo.toml [hunk 3]
```diff
@@ -14,34 +14,38 @@ required-features = ["test-gen"]
...
+merkle_proof.workspace = true
+
 # local
 eth-types.workspace = true
 lightclient-circuits.workspace = true
 reqwest = "0.11.22"
 
 # for test gen binary
-ethereum-consensus = { git = "https://github.com/ralexstokes/ethereum-consensus.git", rev = "f3bff52e9c43866f231ec40c8ab0e34125a8957f", optional = true }
-blst = { version = "0.3.11", default-features = false, features = [ "portable", "no-threads" ], optional = true }
```
Comment (nulltea) [inline preprocessor/Cargo.toml:39]: why use blst by default? hate that it significantly slows build time...

File: preprocessor/src/lib.rs [hunk 4]
```diff
@@ -7,135 +7,51 @@
...
 use lightclient_circuits::witness::{CommitteeUpdateArgs, SyncStepArgs};
 pub use rotation::*;
-use serde::{Deserialize, Serialize};
-use ssz_rs::{Node, Vector};
 pub use step::*;
+use url::Url;
 
-pub async fn get_light_client_update_at_period<S: Spec, C: ClientTypes>(
-    client: &Client<C>,
+pub async fn get_light_client_update_at_period<S: Spec, T: EthSpec>(
```
Comment (nulltea) [inline preprocessor/src/lib.rs]: consider having EthSpec as an associated type in Spec?

File: rust-toolchain
Comment (nulltea) [inline rust-toolchain:1]: why delete?

File: test-utils/src/lib.rs [hunk 5]
```diff
@@ -130,120 +129,118 @@ pub fn read_test_files_and_gen_witness(
...
+    file.read_to_end(&mut data).unwrap();
+
+    let mut decoder = snap::raw::Decoder::new();
+    decoder.decompress_vec(&data).unwrap()
+}
+
+pub fn load_snappy_ssz<T: ssz::Decode>(path: &str) -> Option<T> {
+    let path = Path::new(path);
+    if !path.exists() {
+        // panic!("Path to snappy_ssz file does not exist: {:?} from dir {:?}", path, std::env::current_dir());
```
Comment (nulltea) [inline test-utils/src/lib.rs]: remove or uncomment?

File: prover/src/utils.rs [hunk 6]
```diff
@@ -2,58 +2,70 @@
...
+                })
                 .collect_vec();
 
-            let ssz_root = bootstrap
-                .current_sync_committee
-                .pubkeys
-                .hash_tree_root()
-                .unwrap();
-            println!("SSZ root: {:?}", hex::encode(ssz_root.deref()));
+            // let pubkeys_uncompressed = bootstrap
```
Comment (nulltea) [inline prover/src/utils.rs]: remove?

File: preprocessor/src/step.rs [hunk 7]
```diff
@@ -28,138 +26,177 @@ where
...
+        LightClientFinalityUpdate::Capella(header) => header.attested_header.beacon.state_root,
+        LightClientFinalityUpdate::Deneb(header) => header.attested_header.beacon.state_root,
+    };
 
     let fork_version = client
-        .get_fork(StateId::Root(attested_state_id))
-        .await?
+        .get_beacon_states_fork(StateId::Root(attested_state_id))
+        .await
+        .unwrap()
```
Comment (nulltea) [inline preprocessor/src/step.rs]: use map_err instead

**Review by nulltea — CHANGES_REQUESTED** [review]
Great stuff, thanks! Left some comments

File: preprocessor/src/lib.rs (hunk 4 above)
Comment (ec2) [reply preprocessor/src/lib.rs]: Good idea

File: rust-toolchain
Comment (ec2) [reply rust-toolchain:1]: Mistake. Old toolchain couldnt build axum 0.7.5, so deleted to use latest toolchain. Will re-add with updated toolchain to newest nightly

File: preprocessor/Cargo.toml (hunk 3 above)
Comment (ec2) [reply preprocessor/Cargo.toml:39]: We only needed it for unit_test_gen binary before, but we need it now for the lib as well

File: lightclient-circuits/src/witness/multiproof.rs (hunk 2 above)
Comment (ec2) [reply lightclient-circuits/src/witness/multiproof.rs:181]: I'd say keep it. We will need it for multiproof PR anyways.

**Review by nulltea — APPROVED** [review]
Looks great!

**Pull request merged**